client/src/app/site/deployment-center/deployment-center-setup/step-configure/configure-vsts-build/configure-vsts-build.component.ts

```diff
@@ -65,9 +65,14 @@ export class ConfigureVstsBuildComponent implements OnDestroy {
         .setAsyncValidators([
           VstsValidators.createVstsAccountNameValidator(this.wizard, this._translateService, this._cacheService).bind(this),
         ]);
+      this.wizard.buildSettings.get('vstsProject').setAsyncValidators([]);
+      this.wizard.buildSettings.get('vstsAccount').valueChanges.subscribe(account => {
+        this.wizard.buildSettings.get('vstsProject').setValue(account);
```

Comment:
I think we should fix the project name to some fixed value

Reply:
After discussion with PM, we thought that this should suffice.
But i see your concern, its probably about the allowed names in project and account vary. In this case, do you have some suggestions for the same ?

Reply:
I think the earlier value  was MyFirstProject or something, no? If you have talked to PM - then I am okay with keeping them same - except the invalid char thing (which might not be true)

Reply:
According to online documentation [here](https://docs.microsoft.com/en-us/azure/devops/organizations/settings/naming-restrictions?view=azure-devops)
Not allowed characters in account name: "/ \ [ ] : | < > + = ; ? *
Not allowed characters in project name: / : \ ~ & % ; @ ' " ? < > | # $ * } { , + = [ ]

The characters which are not allowed in project name but are missing from account's list, are actually not allowed in account name as well. I did a manual verification for them. Here is the diff character list:
~ & % @ ' # $ } { ,

Reply:
So basically account name will work as valid project name?

Reply:
Yeah the only thing different was the length, which i have trimmed to max 64 chars

Reply:
{abc}de seem to pass validation for account name in deployment center.. not sure if it's an allowed account name, or it's a bug in Deployment center UX.. I see that you cannot create that account through dev.azure.com. Can you verify?

Reply:
I have added account name regex validation

Reply:
Single character account names are not valid by the given regex.. I'm guessing single character account names are all already taken though.. Also, abc.de is a valid name by given regex, but it'
s not valid as per dev.azure.com

Reply:
Also, abc~ is valid as per regex. Maybe we can not use the negation operator.. we can rather match all invalid cases, and validate to false by code, that'd be easy

Reply:
using static name "MyFirstProject" as suggested, after discussion with Pulkit.